src/ProjectTemplates/Microsoft.Extensions.AI.Templates/src/ChatWithCustomData/ChatWithCustomData-CSharp.Web/Services/IngestedDocument.cs

```diff
@@ -4,23 +4,30 @@ namespace ChatWithCustomData_CSharp.Web.Services;
 
 public class IngestedDocument
 {
-    [VectorStoreRecordKey]
+    private const int VectorDimensions = 2;
+#if (UseAzureAISearch || UseQdrant)
+    private const string VectorDistanceFunction = DistanceFunction.CosineSimilarity;
+#else
+    private const string VectorDistanceFunction = DistanceFunction.CosineDistance;
+#endif
+
+    [VectorStoreKey]
 #if (UseQdrant)
     public required Guid Key { get; set; }
 #else
     public required string Key { get; set; }
 #endif
 
-    [VectorStoreRecordData(IsIndexed = true)]
+    [VectorStoreData(IsIndexed = true)]
     public required string SourceId { get; set; }
 
-    [VectorStoreRecordData]
+    [VectorStoreData]
     public required string DocumentId { get; set; }
 
-    [VectorStoreRecordData]
+    [VectorStoreData]
     public required string DocumentVersion { get; set; }
 
     // The vector is not used but required for some vector databases
-    [VectorStoreRecordVector(2, DistanceFunction = DistanceFunction.CosineSimilarity)]
+    [VectorStoreVector(VectorDimensions, DistanceFunction = VectorDistanceFunction)]
```

Comment:
Should this also be converted from `ReadOnlyMemory<float>` to `string?` (also other instances below)?

Reply:
Ah I see - this isn't actually used, sorry.